lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -302,6 +323,63 @@ prop_2_6_2 (ins, u) =
             ===
         balance u - balance (u `restrictedBy` ins)
 
+{-------------------------------------------------------------------------------
+                        UTxO statistics Properties
+-------------------------------------------------------------------------------}
+
+-- | The 'total' stake in the statistics is the UTxO's balance
+propUtxoTotalIsBalance
+    :: BoundType
+    -> ShowFmt UTxO
+    -> Property
+propUtxoTotalIsBalance bType (ShowFmt utxo) =
+    fromIntegral totalStake == balance utxo
+    & cover 75 (utxo /= mempty) "UTxO /= empty"
+  where
+    UTxOStatistics _ totalStake _ = computeUtxoStatistics bType utxo
+
+-- | The sum of the weighted distribution is greater than the sum of UTxOs
+-- outputs (we distribute the UtxO over buckets with upper-bounds, so everything
+-- in a bucket is lower than its upper-bound).
+propUtxoSumDistribution
+    :: BoundType
+    -> ShowFmt UTxO
+    -> Property
+propUtxoSumDistribution bType (ShowFmt utxo) =
+    sum (upperVal <$> bars) >= fromIntegral (balance utxo)
+    & cover 75 (utxo /= mempty) "UTxO /= empty"
+    & counterexample ("Histogram: " <> pretty bars)
+  where
+    UTxOStatistics bars _ _ = computeUtxoStatistics bType utxo
+    upperVal (HistogramBar k v) = k * v
+
+-- | The distribution is empty if and only if the UTxO is empty
+propUtxoEmptyIsEmpty
+    :: BoundType
+    -> ShowFmt UTxO
+    -> Property
+propUtxoEmptyIsEmpty bType (ShowFmt utxo) =
+    if all isEmpty bars then utxo === mempty else utxo =/= mempty
+    & cover 75 (utxo /= mempty) "UTxO /= empty"
+    & counterexample ("Histogram: " <> pretty bars)
+  where
+    UTxOStatistics bars _ _ = computeUtxoStatistics bType utxo
+    isEmpty (HistogramBar _ v) = v == 0
+
+-- | The sum of the distribution coefficients should is equal to the number of
```

Comment:
> distribution coefficients

This also radically confused me. It appears to be _a thing_, but googling and finding _is the ratio of concentrations of a compound in a mixture of two immiscible phases _ doesn't help me understand this.

I'm not sure what the optimal term is, but maybe something more like "count" or "frequency", or "the sum of the frequencies of all buckets/bins" 🤷‍♂ 

But this is at least the only place where bin counts are called "coefficients"…

Reply:
> <strong>distribution</strong>  
> The action of sharing something out among a number of recipients.

---

> <strong>coefficient</strong>  
> A numerical or constant quantity placed before and multiplying the variable in an algebraic expression (e.g. 4 in 4x y).


Distributions are somewhat "well-known" in mathematics: https://en.wikipedia.org/wiki/Distribution_(mathematics) hence why I coined this term here. Especially in the context of an histogram:

> <strong>histogram</strong>  
> A histogram is an accurate representation of the distribution of numerical data.

Perhaps _bucket's coefficient_ works fine too, like _bucket's count_ or _bucket's frequency_ but they're all really synonym to me :no_mouth: